staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -43,67 +46,98 @@ type FeatureSupportChecker interface {
 	// Supports check if the feature is supported or not by checking internal cache.
 	// By default all calls to this function before calling CheckClient returns false.
 	// Returns true if all endpoints in etcd clients are supporting the feature.
-	Supports(feature storage.Feature) (bool, error)
+	Supports(feature storage.Feature) bool
 	// CheckClient works with etcd client to recalcualte feature support and cache it internally.
 	// All etcd clients should support feature to cause `Supports` return true.
 	// If client A supports and client B doesn't support the feature, the `Supports` will
 	// first return true at client A initializtion and then return false on client B
 	// initialzation, it can flip the support at runtime.
-	CheckClient(ctx context.Context, c client, feature storage.Feature) error
+	CheckClient(ctx context.Context, c client, feature storage.Feature)
 }
 
 type defaultFeatureSupportChecker struct {
 	lock                       sync.Mutex
 	progressNotifySupported    *bool
 	progresNotifyEndpointCache map[string]bool
```

Comment:
With `checkingEndpoint` and each goroutine writing to `progressNotifySupported` directly I don't think you need `progresNotifyEndpointCache` anymore.

Reply:
I asked the same question before but I think we need it because `CheckClient` is called per resource and there is now way to ensure that all calls have been made.

Reply:
In current implementation we may not need it but in future maybe there is some other implementation that may need that. And if we remove this we can abort `checkClient` if one endpoint is not supported.

Reply:
> In current implementation we may not need it but in future maybe there is some other implementation that may need that. And if we remove this we can abort `checkClient` if one endpoint is not supported.

We are writing the code for now, not for the future.

Reply:
If I'm reading the code correctly, newStore is used per resource, so we don't know how many resources we have or when the newStore method will be used, right? If that's the case, it might be that CheckClient will be used with a delay for the same endpoints.

Reply:
Oh, but we don't remove an ep after checking from `f.checkingEndpoint` so yeah, we could get rid of `progresNotifyEndpointCache`

Reply:
> We are writing the code for now, not for the future

Agreed. I'll keep that in mind.